examples/benchmark/benchmark-matmult.cpp

```diff
@@ -32,14 +33,45 @@ static void ggml_graph_compute_helper(std::vector<uint8_t> & buf, ggml_cgraph *
 }
 
 static float tensor_sum_elements(const ggml_tensor * tensor) {
-    double sum = 0;
+    double sum                  = 0;
+    float  floatvalue           = 0;
+    unsigned short shortvalue   = 0;
+
+    if (tensor->type == GGML_TYPE_I8) {
+        for (int j = 0; j < tensor->ne[1]; j++) {
+            for (int k = 0; k < tensor->ne[0]; k++) {
+                sum += ((int8_t *) tensor->data)[j * tensor->ne[0] + k];
+            }
+        }
+    }
+
+    if (tensor->type == GGML_TYPE_F16) {
+        for (int j = 0; j < tensor->ne[1]; j++) {
+            for (int k = 0; k < tensor->ne[0]; k++) {
+                shortvalue = ((unsigned short *) tensor->data)[j * tensor->ne[0] + k];
+                floatvalue = GGML_FP16_TO_FP32(shortvalue);
+                sum        += floatvalue;
+            }
+        }
+    }
+
     if (tensor->type == GGML_TYPE_F32) {
         for (int j = 0; j < tensor->ne[1]; j++) {
             for (int k = 0; k < tensor->ne[0]; k++) {
-                sum += ((float *) tensor->data)[j*tensor->ne[0] + k];
+                sum += ((float *) tensor->data)[j * tensor->ne[0] + k];
             }
         }
     }
+
+    if (ggml_is_quantized(tensor->type)) {
```

Comment:
`GGML_TYPE_I8` is also a quantized type. All `if` conditions in this function does not have `else`, so the sum will be calculated twice for some types.

Reply:
thanks!